### PR TITLE
PORT-9395 | Update sonarqube guide to use search relation

### PR DESCRIPTION
**Pull request by kodjomiles**
# Description

This PR updates the SonarQube analysis to use the new search relation feature.

## Updated docs pages
https://docs.getport.io/build-your-software-catalog/sync-data-to-catalog/code-quality-security/sonarqube/guides/connect-github-pr-with-sonar-analysis

### CODE REVIEW

**Comment by kodjomiles** [comment]
requesting close as this  PR has been moved to the main repository.

File: docs/build-your-software-catalog/sync-data-to-catalog/code-quality-security/sonarqube/guides/connect-github-pr-with-sonar-analysis.md [hunk 1]
```diff
@@ -194,39 +194,127 @@ First, we will need to create a [relation](/build-your-software-catalog/customiz
...
+
+- Go to your [data sources page](https://app.getport.io/settings/data-sources) and click on the GitHub exporter:
+  <img src='/img/guides/githubExporter.png' width='100%' border='1px' />
+
+
+- Replace the mapping YAML with this content:
+
+  ```yaml showLineNumbers
+  resources:
+  - kind: pull-request
```
Comment (hadar-co) [inline docs/build-your-software-catalog/sync-data-to-catalog/code-quality-security/sonarqube/guides/connect-github-pr-with-sonar-analysis.md:227]: fix indentation of this code snippet

**Review by hadar-co — CHANGES_REQUESTED** [review]

File: docs/build-your-software-catalog/sync-data-to-catalog/code-quality-security/sonarqube/guides/connect-github-pr-with-sonar-analysis.md (hunk 1 above)
Comment (kodjomiles) [reply docs/build-your-software-catalog/sync-data-to-catalog/code-quality-security/sonarqube/guides/connect-github-pr-with-sonar-analysis.md]: Done

Comment (kodjomiles) [reply docs/build-your-software-catalog/sync-data-to-catalog/code-quality-security/sonarqube/guides/connect-github-pr-with-sonar-analysis.md]: thank you it was an oversight

File: docs/build-your-software-catalog/sync-data-to-catalog/code-quality-security/sonarqube/guides/connect-github-pr-with-sonar-analysis.md [hunk 2]
```diff
@@ -194,39 +194,127 @@ First, we will need to create a [relation](/build-your-software-catalog/customiz
...
+Go to your [data sources page](https://app.getport.io/settings/data-sources),
+and click on your GitHub Pull Request integration:
+<img src='/img/guides/githubPrIntegrationDataSource.png' border='1px' />
 
 <br/><br/>
 
-Under the `resources` key, the following YAML block to map the pull request entities with analysis. Then click `Save & Resync`:
+## Relation Mapping
+There are two ways
+to establish the relationship between a GitHub `pull-request` entity and a SonarQube `analysis` entity:
```
Comment (PeyGis) [inline docs/build-your-software-catalog/sync-data-to-catalog/code-quality-security/sonarqube/guides/connect-github-pr-with-sonar-analysis.md:208]:
```suggestion
We will explore two ways to establish the relationship between a GitHub `pull-request` entity and a SonarQube `analysis` entity:
```

File: docs/build-your-software-catalog/sync-data-to-catalog/code-quality-security/sonarqube/guides/connect-github-pr-with-sonar-analysis.md [hunk 3]
```diff
@@ -194,39 +194,127 @@ First, we will need to create a [relation](/build-your-software-catalog/customiz
...
 
-<details>
-<summary>Relation mapping (click to expand)</summary>
+### Matching Common Pull Request Title and Branch
+
+The first way is to use the `title` and `branch` properties:
```
Comment (PeyGis) [inline docs/build-your-software-catalog/sync-data-to-catalog/code-quality-security/sonarqube/guides/connect-github-pr-with-sonar-analysis.md:212]:
```suggestion
The first approach is to establish a match using the title and branch properties
```

File: docs/build-your-software-catalog/sync-data-to-catalog/code-quality-security/sonarqube/guides/connect-github-pr-with-sonar-analysis.md [hunk 4]
```diff
@@ -194,39 +194,127 @@ First, we will need to create a [relation](/build-your-software-catalog/customiz
...
+- **Why use `title` and `branch` properties?**
+  The `title` property is common to both GitHub pull requests and SonarQube analyses,
+  making it a reliable identifier for matching related entities.
+  The `branch` property helps specify the exact code changes,
+  ensuring accurate mapping even if there are pull requests with similar titles on different branches.
```
Comment (PeyGis) [inline docs/build-your-software-catalog/sync-data-to-catalog/code-quality-security/sonarqube/guides/connect-github-pr-with-sonar-analysis.md:217]:
```suggestion
  The `branch` property gives information about the source and destination of the the code changes. Combining the title with the branch offers a robust mapping, ensuring that entities are correctly matched in situations where there are similar pull request titles but on different branches.
```

File: docs/build-your-software-catalog/sync-data-to-catalog/code-quality-security/sonarqube/guides/connect-github-pr-with-sonar-analysis.md [hunk 5]
```diff
@@ -194,39 +194,127 @@ First, we will need to create a [relation](/build-your-software-catalog/customiz
...
+                           operator: '"="'
+                           value: .head.ref
+
+  ```
+
+This configuration uses the `title` and `branch` properties to establish a relationship with SonarQube analysis based on matching properties.
+
+### Matching Pull Request Commit SHA
+
+The second way is to use the `commitSha` property:
```
Comment (PeyGis) [inline docs/build-your-software-catalog/sync-data-to-catalog/code-quality-security/sonarqube/guides/connect-github-pr-with-sonar-analysis.md:263]:
```suggestion
The second option is to use the `commitSha` property that is common to both the pull request and SonarQube analysis:
```

**Review by PeyGis — CHANGES_REQUESTED** [review]
left some comments

**Review by hadar-co — APPROVED** [review]

**Review by PeyGis — APPROVED** [review]
LGTM

**Pull request merged**